x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/installation_stats_collector.ts

```diff
@@ -8,15 +8,60 @@ import type { CoreStart, Logger, SavedObjectsClientContract } from '@kbn/core/se
 import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
 import {
   AgentPolicy,
-  PackagePolicy,
   PACKAGE_POLICY_SAVED_OBJECT_TYPE,
+  PackagePolicy,
   SO_SEARCH_LIMIT,
 } from '@kbn/fleet-plugin/common';
 import { agentPolicyService } from '@kbn/fleet-plugin/server/services';
 import type { CloudSecurityInstallationStats } from './types';
 import type { CspServerPluginStart, CspServerPluginStartDeps } from '../../../types';
 import { CLOUD_SECURITY_POSTURE_PACKAGE_NAME } from '../../../../common/constants';
 
+const getEnabledInputStreamVars = (packagePolicy: PackagePolicy) => {
+  const enabledInput = packagePolicy.inputs.find((input) => input.enabled);
+  return enabledInput?.streams[0].vars;
+};
+
+const getCspmTelemetryFields = (packagePolicy: PackagePolicy) => {
+  if (packagePolicy.vars?.posture.value !== 'cspm') return;
+
+  const provider = packagePolicy.vars?.deployment?.value;
+  const inputStreamVars = getEnabledInputStreamVars(packagePolicy);
+  const accountType = inputStreamVars?.[`${provider}.account_type`]?.value;
+
+  if (!accountType) return;
+
+  return {
+    account_type: accountType,
+  };
+};
+
+const getInstalledPackagePolicies = (
+  packagePolicies: PackagePolicy[],
+  agentPolicies: AgentPolicy[]
+) => {
+  const installationStats = packagePolicies.map(
+    (packagePolicy: PackagePolicy): CloudSecurityInstallationStats => {
+      const agentCounts =
+        agentPolicies?.find((agentPolicy) => agentPolicy?.id === packagePolicy.policy_id)?.agents ??
+        0;
+
+      return {
+        package_policy_id: packagePolicy.id,
+        feature: packagePolicy.vars?.posture?.value as string,
+        deployment_mode: packagePolicy.vars?.deployment?.value as string,
+        package_version: packagePolicy.package?.version as string,
+        created_at: packagePolicy.created_at,
+        agent_policy_id: packagePolicy.policy_id,
+        agent_count: agentCounts,
+        cspm: getCspmTelemetryFields(packagePolicy),
```

Comment:
@CohenIdo  why are create another field for `cspm` instead create an `account_type` field?

Reply:
I'm with you on that, I also suggested to use a flat `account_type` field. however, this method was preferred by @kfirpeled. His take is that using a flat field will require us to know the be familiar with the values of this field, and that using a cspm object is adding contextual information.

@CohenIdo also mentioning that you guys had some problem with flat fields in the past.

Ultimately i'd like to do what you guys prefer as you are working much closer to the telemetry than i am.

Let me know if you'd still prefer to have a flat field after hearing the reasoning.

Reply:
+1 @Omolola-Akinleye 

posting here @JordanSh suggestion following our sync:
```typescript
{
	"package_policy_id": "499f20f3-378f-4a1d-8eb9-d600a632f3bb",
	"feature": "cspm",
	"deployment_mode": "aws",
	"package_version": "1.5.0-preview29",
	"created_at": "2023-08-03T11:48:30.171Z",
	"agent_policy_id": "a4f026c0-31f3-11ee-ac73-f96446847dbe",
	"agent_count": 0,
	"account_type": single - account | multy account
}
```

Since we already have information about the cspm from the feature type, there's no need to redundantly include the cspm key. Moreover, using the account_type key will enable us to have a consistent field across different solutions. Currently, for kspm, vul mngm, and gcp, we can populate account_type = 'single-account

Reply:
@CohenIdo the telemetry data after the change:

```json
"installation_stats": [
                {
                  "package_policy_id": "d30d519e-2510-468a-ae49-43eed92f6f5f",
                  "feature": "cspm",
                  "deployment_mode": "aws",
                  "package_version": "1.5.0-preview29",
                  "created_at": "2023-08-03T13:57:08.838Z",
                  "agent_policy_id": "9ed33090-3205-11ee-9fb0-192f024f0a90",
                  "agent_count": 0,
                  "account_type": "organization-account"
                },
                {
                  "package_policy_id": "fc6fb969-8bda-4db3-8f13-d827e496f1fe",
                  "feature": "cspm",
                  "deployment_mode": "gcp",
                  "package_version": "1.5.0-preview29",
                  "created_at": "2023-08-03T12:52:15.458Z",
                  "agent_policy_id": "8ddfad30-31fc-11ee-90a8-45d27b5af585",
                  "agent_count": 0,
                  "account_type": "single-account" <-- if feature is cspm and account_type do not exist on the policy, we assume its 'single-account'
                },
                {
                  "package_policy_id": "499f20f3-378f-4a1d-8eb9-d600a632f3bb",
                  "feature": "cspm",
                  "deployment_mode": "aws",
                  "package_version": "1.5.0-preview29",
                  "created_at": "2023-08-03T11:48:30.171Z",
                  "agent_policy_id": "a4f026c0-31f3-11ee-ac73-f96446847dbe",
                  "agent_count": 0,
                  "account_type": "single-account"
                }
              ]
```

Reply:
On a slightly related topic: does `single-account/organization-account` nomenclature makes sense for Azure and GCP? From what I read in GCP and Azure grouping services might be working differently and naming might be different too, eg. subscription in Azure and project in GCP. Do we see it as a potential problem? As an alternative, we might be a bit more specific in the key names, eg. `aws-single-account` and `aws-organisation-account`? Otherwise if for instance in gcp there will be a third option added, eg. project or smth which doesn't make sense for aws, the data might be more confusing to work with in the future

Reply:
@maxcold that was my initial suggestion as well, i think it would also be the easiest to aggregate over. regarding the current implementation i think its a good balance between what you're suggesting and what kfir and ido wanted initially. we might revisit it in the future but for now i think we should just start with something and come back to it when we actually have the values from gcp and azure

Reply:
Thanks  @JordanSh  Adding the `account-type` field looks good. I agree with @CohenIdo that we can always combine the feature and the account-type fields to see which customers install `cspm` under a multi-organization account or single account.